### PR TITLE
Fix async tests

### PR DESCRIPTION
**Pull request by joelochlann**
While working on #21014 and trying to get the tests to pass, I was getting seriously misled by some dodgy async tests. If you don't actually return the promise containing the `expect` calls, and the assertions fail, Jest will pass the test but give you warnings about unhandled Promise rejections.

It seems this is a fairly common problem, since the difference between returning a Promise and returning undefined is just a matter of adding/removing a pair of `{}`

I've fixed tests that I'm responsible for:
- `ab.spec.js` (also refactored an incomprehensible test that I myself wrote)
- `membership-engagement-banner.spec.js`

and one other that was easy (`breaking.news.spec.js`).

## Others to fix (...or delete)
There are three other tests that I think are actually failing but have dodgy async handling so they're not breaking the build.

I'll leave it up to the @guardian/dotcom-platform team to decide when & whether to do something about those ones

```
(node:8760) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: expect(received).toEqual(expected)
Expected value to equal:
  "-webkit-transform: translate(-100%, 0);"
Received:
  "transform: translate(-100%, 0);"
(node:8760) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:8760) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): Error: expect(received).toEqual(expected)
Expected value to equal:
  "-webkit-transform: translate(-200%, 0);"
Received:
  "transform: translate(-200%, 0);"
(node:8760) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 3): Error: expect(received).toEqual(expected)
Expected value to equal:
  "-webkit-transform: translate(-100%, 0);"
Received:
  "transform: translate(-100%, 0);"
(node:8760) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 4): Error: expect(received).toEqual(expected)
Expected value to equal:
  "-webkit-transform: translate(-000%, 0);"
Received:
  "transform: translate(-000%, 0);"
(node:8760) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 5): Error: expect(received).toEqual(expected)
Expected value to equal:
  "-webkit-transform: translate(-300%, 0);"
Received:
  "transform: translate(-300%, 0);"
(node:8760) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 6): Error: expect(received).toEqual(expected)
Expected value to equal:
  "-webkit-transform: translate(-100%, 0);"
Received:
  "transform: translate(-100%, 0);"
(node:8760) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 7): Error: expect(received).toEqual(expected)
Expected value to equal:
  "-webkit-transform: translate(-200%, 0);"
Received:
  "transform: translate(-200%, 0);"
(node:8760) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 8): Error: expect(received).toEqual(expected)
Expected value to equal:
  "-webkit-transform: translate(-000%, 0);"
Received:
  "transform: translate(-000%, 0);"
 PASS  static/src/javascripts/projects/commercial/modules/hosted/onward-journey-carousel.spec.js


(node:8758) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: Cannot read property 'id' of undefined
(node:8758) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:8758) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): TypeError: Cannot read property 'id' of undefined
(node:8758) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 3): TypeError: Cannot read property 'id' of undefined
(node:8758) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 4): TypeError: Cannot read property 'id' of undefined
(node:8758) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 5): TypeError: Cannot read property 'id' of undefined
(node:8758) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 6): TypeError: Cannot read property 'html' of undefined
(node:8758) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 7): TypeError: Cannot read property 'html' of undefined
(node:8758) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 8): TypeError: Cannot read property 'id' of undefined
(node:8758) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 9): TypeError: Cannot read property 'html' of undefined
(node:8758) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 10): TypeError: Cannot read property 'html' of undefined
(node:8758) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 11): TypeError: Cannot read property 'html' of undefined
 PASS  static/src/javascripts/projects/facia/modules/onwards/weather.spec.js


(node:8759) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): TypeError: labelEl.querySelector is not a function
(node:8759) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:8759) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 4): TypeError: labelEl.querySelector is not a function
(node:8759) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 6): TypeError: labelEl.querySelector is not a function
(node:8759) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 8): TypeError: Cannot read property 'add' of undefined
(node:8759) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 10): TypeError: Cannot read property 'add' of undefined
 PASS  static/src/javascripts/projects/common/modules/identity/modules/switch.spec.js
```

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/experiments/ab.js [hunk 1]
```diff
@@ -61,7 +61,7 @@ export const getSynchronousTestsToRun = memoize(
     }
 );
 
-export const getAynschronousTestsToRun = (): Promise<
+export const getAsyncTestsToRun = (): Promise<
```
Comment (joelochlann) [inline static/src/javascripts/projects/common/modules/experiments/ab.js]: this is just fixing a typo of mine. seems like `async` is harder to mis-spell...

**Review by tomrf1 — APPROVED** [review]
😬

**Comment by SiAdcock** [comment]
Great, thanks @joelochlann !

There's a [Trello card](https://trello.com/c/HWQrPKLX/113-fix-unhandledpromiserejectionwarnings-in-js-tests) that's been on our backlog for a year, so thanks for making a start on it!

**Comment by joelochlann** [comment]
haha!

I've got to say, it's so rare to find a case where the success/failure of a unit test had anything to do with the correctness of the real-world functionality, as opposed to correctness of the mocking/testing code...

**Pull request merged**